src/sentry/tasks/data_export.py

```diff
@@ -155,9 +156,71 @@ def attach_eventuser(items):
     return file_name
 
 
-def process_discover(data_export, file):
-    # TODO(Leander): Implement processing for Discover
-    raise NotImplementedError("Discover processing has not been implemented yet")
+def process_discover(data_export, file, limit):
+    """
+    Convert the discovery query to a CSV, writing it to the provided file.
+    Returns the suggested file name.
+    (Adapted from 'src/sentry/api/endpoints/organization_events.py')
+    """
+    payload = data_export.query_info
+
+    start, end = get_date_range_from_params(payload)
+    try:
+        project = Project.objects.get(id=payload["project"])
```

Comment:
Discover queries can often target more than a single project. Will exports only work for one project?

Reply:
Thanks! these are the type of things i forget to check, im working on a better solution now

Reply:
Multi-project queries are supported in the latest changes!